impl/src/main/java/com/sun/faces/config/manager/tasks/ProvideMetadataToAnnotationScanTask.java

```diff
@@ -72,10 +74,12 @@ private void initializeIvars(Set<Class<?>> annotatedSet) {
                          * annotatedSet because the faces-config.xml that owns it has metadata-complete="true".
                          */
                         ArrayList<Class<?>> toRemove = new ArrayList<>(1);
-                        String sourceURIString = sourceURI.toString();
+                        String facesConfigURIString = facesConfigURI.toString();
                         if (annotatedSet != null) {
                             for (Class<?> clazz : annotatedSet) {
-                                if (sourceURIString.contains(clazz.getProtectionDomain().getCodeSource().getLocation().toString())) {
+                                String jarURIString = CURRENT_RESOURCE_IN_JAR_PATTERN.split(clazz.getResource(".").toString(), 2)[0];
```

Comment:
Not seeing a test here that validates this change over?

Reply:
Added integration test: https://github.com/eclipse-ee4j/mojarra/pull/5519/commits/1ff67143b9f9ab5dba14d1b8932a982b8dc47399